Sources/SwiftFusion/Inference/GenericFactor.swift

```diff
@@ -78,16 +78,12 @@ extension Tuple: VariableTuple where Tail: VariableTuple {
   ) -> R {
     return variableAssignments
       .contiguousStorage[ObjectIdentifier(Head.self)].unsafelyUnwrapped
-      .withUnsafeMutableRawBufferPointer { headBuffer in
-        let headBase: UnsafePointer<Head>
-        if let rawHeadBase = headBuffer.baseAddress {
-          headBase = UnsafePointer(rawHeadBase.assumingMemoryBound(to: Head.self))
-        } else {
-          // An invalid pointer is okay because we know it'll never get dereferenced.
-          headBase = UnsafePointer<Head>(bitPattern: -1).unsafelyUnwrapped
-        }
+      .withUnsafeRawPointerToElements { headBase in
         return Tail.withVariableBufferBaseUnsafePointers(variableAssignments) { tailBase in
-          return body(UnsafePointers(head: headBase, tail: tailBase))
+          return body(UnsafePointers(
```

Comment:
nit: if you run swift-format on this you'll get a newline after the first paren.  The rule is, if you have an opening (,{,[ delimiter on a line and can't fit the matching close delimiter on the same line, the open delimiter is the last thing on the line, except for comments and closure signatures. 
```swift
{ x, y in
    ... 
}
```

Reply:
done